### PR TITLE
Interview pairing guidance

### PR DESCRIPTION
**Pull request by craigjbass**

### CODE REVIEW

File: guides/hiring/pairing.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+- Make sure the assessor knows what to do (putting them at ease too)
+
+## Guidance
+
+1. Ask if they’ve done any pair programming before
+2. Ask if they’re familiar with any styles of pair programming. e.g. Ping-Pong, Driver-Navigator.
+3. Agree on a pairing style by explaining the rules.
+4. Set the expectation that you will be pairing with them, as an active participant.
+5. Ask if they are familiar with code katas.
+6. If they are familiar, ask them about their favorities. Make a judgement call on the complexity (e.g. katas with a grid can be too much for 30 mins)
```
Comment (dushan-madetech) [inline guides/hiring/pairing.md]: think you need to correct favorities to favourites

File: guides/hiring/pairing.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+13. Afterwards, thank them for pairing and shake their hand, if culturally appropriate. This goes a long way to project our culture of respect, trust and knowledge sharing. 
+
+## What we are looking for
+
+**Knowledge & Skill**
+
+- Ability to code (tests and production code).
+- (If using their own machine), ability to create a new project and execute code.
+- Pair Programming.
+- Good approach to problem-solving (TDD or otherwise).
```
Comment (StevenLeighton21) [inline guides/hiring/pairing.md]: Guidance around otherwise? Can I suggest a diagram if we're not TDDing?

Comment (craigjbass) [reply guides/hiring/pairing.md]: Yea if people are struggling to take a TDD approach, I wont be trying to teach them in 30 mins - instead I look for other coping mechanisms

- Pen & Paper
- Diagrams
- Writing comments with thought process
- REPL-driven development

Non-exhaustive - want to let the interviewer judge the sanity of the approach

File: guides/hiring/pairing.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+3. Agree on a pairing style by explaining the rules.
+4. Set the expectation that you will be pairing with them, as an active participant.
+5. Ask if they are familiar with code katas.
+6. If they are familiar, ask them about their favourites. Make a judgement call on the complexity (e.g. katas with a grid can be too much for 30 mins)
+  - it shouldn't matter if you've tackled the specific kata before or not.
+7. If they are not familiar, or they don’t have a favourite, take them to learn.madetech.com and discuss the options.
+  - tennis or roman numerals are good for the time available.
+  - don't do refactoring katas because they solely focus on refactoring skills. 
+9. Set the expectation that the kata doesn’t need to be finished.
+10. The candidate or you can write the first test,
```
Comment (StevenLeighton21) [inline guides/hiring/pairing.md]: `The candidate or you can write the first test.`

File: guides/hiring/pairing.md (hunk 2 above)
Comment (StevenLeighton21) [reply guides/hiring/pairing.md]: Dealer's choice, then :)

**Comment by StevenLeighton21** [comment]
General comment - I feel consistency in process also includes consistency in expectations set before the interview (specifically, this is to do with experience bringing a personal laptop vs using a pre-prepared laptop in a language agreed beforehand).

I'm not sure it is for this document, but wanted to prompt the discussion.

File: guides/hiring/pairing.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+## Guidance
+
+1. Ask if they’ve done any pair programming before
+2. Ask if they’re familiar with any styles of pair programming. e.g. Ping-Pong, Driver-Navigator.
+3. Agree on a pairing style by explaining the rules.
+4. Set the expectation that you will be pairing with them, as an active participant.
+5. Ask if they are familiar with code katas.
+6. If they are familiar, ask them about their favourites. Make a judgement call on the complexity (e.g. katas with a grid can be too much for 30 mins)
+  - it shouldn't matter if you've tackled the specific kata before or not.
```
Comment (kjdchapman) [inline guides/hiring/pairing.md]: These bullets aren't working properly in the generated markdown.

File: guides/hiring/pairing.md (hunk 3 above)
Comment (kjdchapman) [inline guides/hiring/pairing.md]:
```suggestion
10. You're free to choose who writes the first test - you or the candidate.
```

**Comment by kjdchapman** [comment]
I feel like I could take a language pass at this - I've caught the simplicity bug from some lovely GDS content people - but I don't know:
- if a github review is the right format for it
- if it's a valuable thing to spend time on (it'd probably take ~30 mins)

What do you think, @craigjbass ?

**Pull request merged**